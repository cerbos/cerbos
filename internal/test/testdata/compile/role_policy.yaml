# yaml-language-server: $schema=../.jsonschema/CompileTestCase.schema.json
---
mainDef: "role_policies/acme_admin.yaml"
