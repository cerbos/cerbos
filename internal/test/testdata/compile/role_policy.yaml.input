-- role_policies/acme_admin.yaml --
---
apiVersion: api.cerbos.dev/v1
rolePolicy:
  role: leave_request
  scope: "acme"
  rules:
    - resource: acme_admin
      permissibleActions: ['*']
    - resource: expenses_report
      permissibleActions:
        - view
        - create
        - delete
