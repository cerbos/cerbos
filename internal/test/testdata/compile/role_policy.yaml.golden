{
  "fqn": "cerbos.role.leave_request/acme",
  "rolePolicy": {
    "meta": {
      "fqn": "cerbos.role.leave_request/acme",
      "sourceAttributes": {
        "role.leave_request/acme": {
          "attributes": {
            "source": "role_policies/acme_admin.yaml"
          }
        }
      }
    },
    "role": "leave_request",
    "scope": "acme",
    "resources": {
      "acme_admin": {
        "actions": {
          "*": {}
        }
      },
      "expenses_report": {
        "actions": {
          "create": {},
          "delete": {},
          "view": {}
        }
      }
    }
  },
  "compilerVersion": 1
}
